src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -585,7 +601,17 @@ int ITupleInternal.GetHashCode(IEqualityComparer comparer)
         /// </remarks>
         public override string ToString()
         {
-            return "(" + Item1 + ", " + Item2 + ")";
+            return "(" + ToStringEnd();
+        }
+
+        string ITupleInternal.ToStringEnd()
+        {
+            return ToStringEnd();
+        }
+
+        private string ToStringEnd()
+        {
+            return Item1 + ", " + Item2 + ")";
```

Comment:
This pattern is a bit unfortunate, as it means that the common case of calling ToString on a ValueTuple of arity 1-7 is being penalized for the less common case of calling ToString on a ValueTuple of arity greater 7.

There are a variety of alternatives that could avoid the extra string allocation here.  The most obvious is you could just duplicate the implementation across the two methods, e.g.
```C#
public override string ToString() => "(" + Item1 + ", " + Item2 + ")";
string ITupleInternal.ToStringEnd() => Item1 + ", " + Item2 + ")";
```
If you wanted to avoid such duplication, you could incur some small extra cost just for the ValueTuple of arity greater than 7 by not having ITupleInternal.ToStringEnd at all, and in ValueTuple`8's ToString implementation using ```rest.ToString().Substring(1)``` instead of ```rest.ToStringEnd();```.

There are of course other alternatives as well.

Regardless of what approach you take, I'd suggest that rather than just doing ```+ ItemN + ```, you do ```+ ItemN.ToString() +```.  With the current code, the compiler is going to use the string.Concat overloads that take objects or object[]s.  This means that for value type items, they're going to get boxed and then have ToString called on them, rather than just having ToString called on them to begin with.  Further, for most of the ToString implementations, the object[] overload is going to be used, so the compiler is going to allocate an object[] for the params, and then the string.Concat implementation is going to allocate a string[] to store the results of the ToStrings before creating the resulting string.  If you instead use ToString on each item, you'll use the string.Concat overload that takes strings or string[]s, and you'll avoid both the box per item as well as the object[] allocation.